tooltester/src/main/java/io/dockstore/tooltester/runWorkflow/WorkflowRunner.java

```diff
@@ -339,20 +394,90 @@ private Execution.ExecutionStatusEnum getExecutionStatus() {
         }
     }
 
+    private List<String> getListOfEcsTasks() {
+        ListTaskDefinitionsRequest request = ListTaskDefinitionsRequest.builder().build();
+        ListTaskDefinitionsResponse response = ecsClient.listTaskDefinitions(request);
+        return response.taskDefinitionArns();
+    }
+
+
+    private void setTaskFamily(final List<String> listOfEcsTasksBeforeNewOneWasAdded) {
```

Comment:
**The Challenge with getting metrics for WDL workflows**

An ECS task is created for WDL workflows, but its naming is a bit confusing to me, for example, this workflow creates this [ECS](https://qa.dockstore.org/workflows/github.com/gatk-workflows/seq-format-conversion/BAM-to-Unmapped-BAM:3.0.0?tab=tools) task,
```
cromwell_broadinstitute_gatk_latest_3aba5ebc355961fe2321e9473f33a16863c67b16
```
I believe the name has something to do with the docker image being used. I haven't found a reliable way to determine the docker image and try and determine what task is currently running.

Now the each ECS task associated with a WDL will remain active for several days after it is run, and if the same workflow is run again, and the ECS task is still active, then the AGC will simply use the active ECS task. This is different to what CWL does, as it **always** creates a new task.

This also means that for WDL tasks I cannot just check to see if a new task has been created to determine the `taskDefinitionFamily` for metrics.

To get around this, I have found (only using the console for now) that if I deregister an ECS WDL task after it has finished running, and then I run the workflow again, a new ECS task will be created for the WDL workflow, meaning that I can use the same approach used for CWL workflows.

What I would like to do, is to add some code at the end of each WDL run that deregisters the associated ECS task when the WDL run is over, this will guarantee that a new ECS task will be created, should the WDL task be run again.

I am slightly cautious about this, as I do not want to accidentally deregister an important ECS task not associated with the tooltester.

Reply:
Is it okay to attempt to do this?

Reply:
Yeah, I think we'll keep this well separated from qa, staging, prod though :laughing: 
But I think it's worth a try. Good detective work, this is more intense than I had expected

Reply:
For this PR, I think I might display a message at the end of the run instructing the user to de-register the task (and the name of the job), that way the user will notice if anything fishy is going on.

Then once a new account that is very separate from qa, staging, and prod is created, I'll have it done automatically.

Reply:
Well, dev is already well separated. I was more joking that if this autokills tasks, we'd never move it into our main account (which is totally fine)

Reply:
Okay, so I'm good to attempt to do this? Just looking at the dev account, the only tasks that I could accidentally kill (that aren't mine), are:

[Dockstore-1124a-task-definition](https://us-east-1.console.aws.amazon.com/ecs/v2/task-definitions/Dockstore-1124a-task-definition?status=ACTIVE?region=us-east-1)	
	[Dockstore-Definition-fargate-dockstore](https://us-east-1.console.aws.amazon.com/ecs/v2/task-definitions/Dockstore-Definition-fargate-dockstore?status=ACTIVE?region=us-east-1)	
	[JumpInstanceCdkStack-jump-server-task-definition](https://us-east-1.console.aws.amazon.com/ecs/v2/task-definitions/JumpInstanceCdkStack-jump-server-task-definition?status=ACTIVE?region=us-east-1)	
	[cromwell_broadinstitute_gatk_latest_3aba5ebc355961fe2321e9473f33a16863c67b16](https://us-east-1.console.aws.amazon.com/ecs/v2/task-definitions/cromwell_broadinstitute_gatk_latest_3aba5ebc355961fe2321e9473f33a16863c67b16?status=ACTIVE?region=us-east-1)	
	[fargate-dev-dockstore-task-definition](https://us-east-1.console.aws.amazon.com/ecs/v2/task-definitions/fargate-dev-dockstore-task-definition?status=ACTIVE?region=us-east-1)	
	[fargate-dockstore-task-definition](https://us-east-1.console.aws.amazon.com/ecs/v2/task-definitions/fargate-dockstore-task-definition?status=ACTIVE?region=us-east-1)	
	[fargate-dockstore2-task-definition](https://us-east-1.console.aws.amazon.com/ecs/v2/task-definitions/fargate-dockstore2-task-definition?status=ACTIVE?region=us-east-1)	
	[kathysDockstore-1-13-0-task-definition](https://us-east-1.console.aws.amazon.com/ecs/v2/task-definitions/kathysDockstore-1-13-0-task-definition?status=ACTIVE?region=us-east-1)	
	[kathysJump-task-definition](https://us-east-1.console.aws.amazon.com/ecs/v2/task-definitions/kathysJump-task-definition?status=ACTIVE?region=us-east-1)